Code/Framework/AzCore/AzCore/std/ranges/filter_view.h

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/ranges/all_view.h>
+#include <AzCore/std/ranges/ranges_adaptor.h>
+#include <AzCore/std/ranges/ranges_algorithm.h>
+
+namespace AZStd::ranges
+{
+    template<class View, class Pred, class = enable_if_t<conjunction_v<
+        bool_constant<input_range<View>>,
+        bool_constant<view<View>>,
+        bool_constant<copy_constructible<Pred>>,
+        is_object<Pred>,
+        bool_constant<indirect_unary_predicate<Pred, iterator_t<View>>>
+        >
+        >>
+    class filter_view;
+
+
+    // views::filter customization point
+    namespace views
+    {
+       namespace Internal
+        {
+            struct filter_fn
+                : Internal::range_adaptor_closure<filter_fn>
+            {
+                template <class View, class Pred, class = enable_if_t<conjunction_v<
+                    bool_constant<viewable_range<View>>
+                    >>>
+                constexpr auto operator()(View&& view, Pred&& func) const
+                {
+                    return filter_view(AZStd::forward<View>(view), AZStd::forward<Pred>(func));
+                }
+
+                // Create a range_adaptor forwarder which binds the pattern for later
+                template <class Pred, class = enable_if_t<constructible_from<decay_t<Pred>, Pred>>>
+                constexpr auto operator()(Pred&& func) const
+                {
+                    return range_adaptor_closure_forwarder(
+                        [*this, func = AZStd::forward<Pred>(func)](auto&& view) mutable
+                        {
+                            // Pattern needs to be decayed in case it's type is array
+                            return (*this)(AZStd::forward<decltype(view)>(view), AZStd::forward<Pred>(func));
+                        }
+                    );
+                }
+            };
+        }
+        inline namespace customization_point_object
+        {
+            constexpr Internal::filter_fn filter{};
+        }
+    }
+
+    template<class View, class Pred, class>
+    class filter_view
+        : public view_interface<filter_view<View, Pred>>
+    {
+        struct iterator;
+        struct sentinel;
+
+    public:
+        template <bool Enable = default_initializable<View> && default_initializable<Pred>,
+            class = enable_if_t<Enable>>
+        filter_view() {}
+
+         constexpr filter_view(View base, Pred func)
+            : m_base(AZStd::move(base))
+            , m_func(AZStd::move(func))
+        {
+        }
+
+        template <bool Enable = copy_constructible<View>, class = enable_if_t<Enable>>
+        constexpr View base() const&
+        {
+            return m_base;
+        }
+        constexpr View base() &&
```

Comment:
uber nit: Missing line break here for consistency

Reply:
Actually this was purposeful to "group" the `base` functions together as they have the same name.